src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -391,6 +392,7 @@ class CDCL(private val solverType: SolverType = SolverType.INCREMENTAL) : Increm
         setStatus(v, VarStatus.UNDEFINED)
         vars[v].clause = null
         vars[v].level = -1
+        variableSelector.backTrack(v)
```

Comment:
Note that `undefineVariable` method is (if I understand correctly) unsafe to call without also modifying the trail. So consider making a method `backtrack` (general name) `cancelUntil` (minisat name) for simultaneous backtracking, trail shortening and corresponding variable reset. Overall, my point is that you (probably) do not need to call this method outside of `backtrack`, so I suggest not to pollute the interface. If this is not your case, feel free to leave it as it is!